File: src/main/java/org/jdc/template/ui/fragment/SettingsFragment.java
package org.jdc.template.ui.fragment;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v7.preference.Preference;
import android.support.v7.preference.PreferenceFragmentCompat;

import org.jdc.template.R;
import org.jdc.template.inject.Injector;

public class SettingsFragment extends PreferenceFragmentCompat implements SharedPreferences.OnSharedPreferenceChangeListener,
        Preference.OnPreferenceClickListener {

    public static SettingsFragment newInstance() {
        return new SettingsFragment();
    }

    public SettingsFragment() {
        Injector.get().inject(this);
    }

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        addPreferencesFromResource(R.xml.preferences);
    }

    @Override
    public boolean onPreferenceClick(Preference preference) {
        return false;
    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {

    }
}

File: app/src/main/kotlin/org/jdc/template/ui/fragment/SettingsFragment.kt
package org.jdc.template.ui.fragment

import android.os.Bundle
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.findNavController
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import dagger.hilt.android.AndroidEntryPoint
import org.jdc.template.R
import org.jdc.template.model.prefs.DisplayThemeType
import org.jdc.template.util.enumValueOfOrNull

@AndroidEntryPoint
class SettingsFragment : PreferenceFragmentCompat() {

    private val viewModel: SettingsViewModel by viewModels()

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        addPreferencesFromResource(R.xml.preferences)

        setupPreferenceListeners()
    }

    private fun setupPreferenceListeners() {
        findPreference<Preference>("displayThemeType")?.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _, newValue ->
            when (newValue) {
                is String -> {
                    enumValueOfOrNull<DisplayThemeType>(newValue)?.let { theme ->
                        viewModel.setTheme(theme)
                    }
                }
            }
            true
        }

        findPreference<Preference>("workManagerStatusButton")?.setOnPreferenceClickListener {
            findNavController().navigate(SettingsFragmentDirections.actionToWorkManagerFragment())
            true
        }
    }
}
